crates/verify/src/etherscan/mod.rs

```diff
@@ -252,37 +244,52 @@ impl EtherscanVerificationProvider {
     /// Create an Etherscan client.
     pub(crate) fn client(
         &self,
-        chain: Chain,
-        verifier_type: &VerificationProviderType,
-        verifier_url: Option<&str>,
-        etherscan_key: Option<&str>,
+        etherscan_opts: &EtherscanOpts,
+        verifier_args: &VerifierArgs,
         config: &Config,
     ) -> Result<Client> {
+        let chain = etherscan_opts.chain.unwrap_or_default();
+        let etherscan_key = etherscan_opts.key();
+        let verifier_type = &verifier_args.verifier;
+        let verifier_url = verifier_args.verifier_url.as_deref();
+
+        // Verifier is etherscan if explicitly set or if no verifier set (default sourcify) but
+        // API key passed.
+        let is_etherscan = verifier_type.is_etherscan() ||
+            (verifier_type.is_sourcify() && etherscan_key.is_some());
         let etherscan_config = config.get_etherscan_config_with_chain(Some(chain))?;
 
+        let api_version = verifier_args.verifier_api_version.unwrap_or_else(|| {
+            if is_etherscan {
+                etherscan_config.as_ref().map(|c| c.api_version).unwrap_or_default()
+            } else {
+                EtherscanApiVersion::V1
```

Comment:
I see, this is for non v2 compat verifiers?

Reply:
yes, they'll default to v1, we can gradually move that to default v2 as they upgrade